### PR TITLE
Extract helper methods for form input_html in auth/ views

### PR DESCRIPTION
**Pull request by mjankowski**

### CODE REVIEW

File: app/helpers/auth/registrations_helper.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Auth::RegistrationsHelper
+  def registrations_new_confirm_password_input_html
```
Comment (ClearlyClaire) [inline app/helpers/auth/registrations_helper.rb:4]: If this is to be extracted to a new helper, I think it should have a less confusing name explicitly referencing this is a honeypot field.

**Review by ClearlyClaire — COMMENTED** [review]
I am honestly not sure how much of an improvement that is. This does make the HAML files a little bit more concise, but this splits the logic across more files, moving parts to single-use methods.

File: app/helpers/auth/registrations_helper.rb (hunk 1 above)
Comment (mjankowski) [reply app/helpers/auth/registrations_helper.rb]: Added honeypot to naming.

**Comment by mjankowski** [comment]
> I am honestly not sure how much of an improvement that is. This does make the HAML files a little bit more concise, but this splits the logic across more files, moving parts to single-use methods.

Another option is to use the multi-line haml style (trailing pipes), but in my opinion this looks way worse. In addition to the method extraction naming clarity here, another benefit is that the spec coverage report becomes more useful because you can see that the helper methods all get called, instead of being swallowed up into the one longer line (where you don't know which parts of it were actually exercised).

I think the "sprawling collection of single use helper methods" critique is valid ... and suspect there could be opportunity to consolidate this general approach across multiple views/helpers down into something cleaner and more general. For example, if you look just at the current batch of ~5 related PRs from me doing this sort of thing, these are only touching the areas where haml-lint was complaining ... I suspect that if I kept expanding the refactor across the full /admin/ area, /settings/ area, etc - we could refactor all that down to fewer methods across fewer helpers. This might bring more overall consistency and a smaller collection of general purpose methods instead of more spread out single use ones.

**Comment by mjankowski** [comment]
Upon review, I agree that the refactor here is not enough of a gain for the sort of sprawling helper file created. Closing this - may open another which adds a narrow helper for JUST the common elements of the various "new password" fields here.

**Pull request closed without merging**